src/chunked/decoder.rs

```diff
@@ -583,20 +580,30 @@ mod tests {
                     .as_bytes(),
             );
             let (s, r) = async_std::sync::channel(1);
-            let sender = TrailersSender::new(s);
+            let sender = Sender::new(s);
             let mut decoder = ChunkedDecoder::new(input, sender);
 
             let mut output = String::new();
             decoder.read_to_string(&mut output).await.unwrap();
             assert_eq!(output, "MozillaDeveloperNetwork");
 
-            let trailer = r.recv().await.unwrap().unwrap();
+            let trailers = r.recv().await.unwrap();
+            let as_vec = trailers.iter().collect::<Vec<_>>();
+
+            assert_eq!(as_vec.len(), 1);
+            let (header_name, header_values) = as_vec[0];
+            assert_eq!(
+                header_name,
+                &"Expires"
+                    .parse::<http_types::headers::HeaderName>()
+                    .unwrap()
+            );
+            assert_eq!(header_values.iter().collect::<Vec<_>>().len(), 1);
```

Comment:
I believe we should be able to do much simpler assertions here

Reply:
I _think_ this is resolved, but I also didn't do a particularly thorough reworking of any of these tests, just got them to compile and pass